hazelcast/src/main/java/com/hazelcast/client/impl/protocol/util/ClientMessageDecoder.java

```diff
@@ -44,10 +50,22 @@
 
     private final Connection connection;
     private final Long2ObjectHashMap<ClientMessageReader> builderBySessionIdMap = new Long2ObjectHashMap<>();
-    private ClientMessageReader activeReader = new ClientMessageReader();
+    private ClientMessageReader activeReader;
 
-    public ClientMessageDecoder(Connection connection, Consumer<ClientMessage> dst) {
+    private boolean clientIsTrusted;
```

Comment:
should this be volatile?

Reply:
I'm expecting it running on a single thread. Isn't it the case?
The other field (`activeReader`) and fields within the `ClientMessageReader` class are not volatile.